app/models/tag.rb

```diff
@@ -817,6 +817,8 @@ def transfer_or_remove_associations
       self.sub_taggings.destroy_all
       self.meta_taggings.destroy_all
     end
+
+    refresh_autocomplete
```

Comment:
Would this plus the above result in refreshing autocomplete for tags twice?

Reply:
Oops, sorry, wrong code: This does absolutely nothing.

That's the one that was supposed to be in the PR: https://github.com/otwcode/otwarchive/pull/4635/commits/f190e82f3b27da02693c0d26caa7b9fa7b55f7f9

And yes, the idea is to refresh the autocomplete for all children at the end, in a separate async task for good measure, even if it means doing it twice potentially.

I tested it through the technique of "commenting the after_commit in common_tagging". Not sure how to write a clean test for that case as I'm actually not sure how it's even possible for the issue to arise in real life.

Reply:
Whoops! In theory, we could do something with mocks/spies, but at that point the argument in favour of a unit test starts to get a little bit weaker (maybe still nice for documentation purposes? but even then tests with reliance on mocking can be tricky).

To take a slightly different angle on my original question: do we need https://github.com/otwcode/otwarchive/pull/4635/files#diff-b361dece27e8fd54e644111ffe427e4bc584369b48b920c781c16803ed65cd4bR24 alongside the code from https://github.com/otwcode/otwarchive/commit/f190e82f3b27da02693c0d26caa7b9fa7b55f7f9? It seems to me that the latter would be the most comprehensive, but it's possible I forgot something about autocomplete